src/domain/challenge/hub/hub.resolver.fields.ts

```diff
@@ -57,15 +66,31 @@ export class HubResolverFields {
     return await this.hubService.getCommunityInNameableScope(ID, hub);
   }
 
+  @AuthorizationAgentPrivilege(AuthorizationPrivilege.READ)
   @ResolveField('context', () => IContext, {
     nullable: true,
     description: 'The context for the hub.',
   })
+  @UseGuards(GraphqlGuard)
   @Profiling.api
   async context(@Parent() hub: Hub) {
     return await this.hubService.getContext(hub);
   }
 
+  @AuthorizationAgentPrivilege(AuthorizationPrivilege.READ)
+  @UseGuards(GraphqlGuard)
+  @ResolveField('collaboration', () => ICollaboration, {
+    nullable: true,
+    description: 'The collaboration for the Hub.',
+  })
+  @Profiling.api
+  async collaboration(@Parent() hub: Hub) {
+    return await this.baseChallengeService.getCollaboration(
```

Comment:
I'd rather see a call to get the collaboration from the hubService, you have the hub repo there.